toDoApp/app/src/main/java/hieunguyen/com/todoapp/MyAdapter.java

```diff
@@ -0,0 +1,83 @@
+package hieunguyen.com.todoapp;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+
+import java.util.ArrayList;
+
+/**
+ * Created by hieunguyen on 9/11/15.
+ */
+public class MyAdapter extends ArrayAdapter<ToDoItem> {
+
+    private ArrayList<ToDoItem> itemList;
+    private MainService dbService;
+    private LayoutInflater myInflater = null;
+
+    public MyAdapter(Context context, MainService service) {
+        super(context, R.layout.target_item, service.getDataFromDB());
```

Comment:
So this doesn't break the code because you are Overriding getCount and getView, but
the default ArrayAdapter expects that the list that is passed in the constructor will be the list that contains the data it needs to look at, and when notifyDataSetChanged() is called it will check the contents of that list. However, here itemList and the ArrayAdapter constructor are pointing to different lists.